### ISSUE TITLE
`noqa` comments on long lines incorrectly moved to its own line

### ISSUE THREAD

**Issue by tunetheweb** [sqlfluff/sqlfluff#4248]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Run `fix` on the following SQL using `ansi` dialect using default config:

```sql
SELECT
    col1,
    col2,
    col3
FROM
    really_really_really_really_really_really_long_schema_name.TABLE1 -- noqa: L014
```

The `noqa` is moved to the new line, where it no longer effects the line it needs to effect.

### Expected Behaviour

I think there’s a few cases:

1. if `ignore_comment_clauses` and/or `ignore_comment_lines` is set and the comment is the only reason the line length is exceeded, then to me it should not be a violation.
2. If those are not set, or it’s exceeded anyway I think this should be unfixable, as it’s not clear how the line should be split and where the `noqa` should go. The alternative is that we add `noqa` to all new lines, but that seems like it might be overuse of `noqa`.

### Observed Behaviour

SQL is incorrectly fixed and then flags on the rule that should be `noqa`d in subsequent runs.

### How to reproduce

As per above

### Dialect

All, but above example works in ANSI.

### Version

Main branch for upcoming version 2.

### Configuration

Default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
